starter/src/main/java/com/starter/fullstack/dao/InventoryDAO.java

```diff
@@ -82,7 +81,10 @@ public Optional<Inventory> update(String id, Inventory inventory) {
    * @return Deleted Inventory.
    */
   public Optional<Inventory> delete(String id) {
-    // TODO
-    return Optional.empty();
+    Inventory inventory = mongoTemplate.findById(id, Inventory.class);
+    if (inventory != null) {
+      mongoTemplate.remove(inventory);
+    }
```

Comment:
Although searching for it then removing it works, it's not the most efficient way to do it. See if you can find another method in the MongoTemplate javadocs that would be more appropriate.

Reply:
Also, pay attention to the return types when looking at javadoc. Knowing what certain things returns can help you in the long run.